org.openscenegraph.android/jni/JNIosgDB.cpp

```diff
@@ -85,4 +95,40 @@ JNIEXPORT jlong JNICALL Java_org_openscenegraph_osg_db_ReadFile_nativeReadNodeFi
 
 }
 
+JNIEXPORT jlong JNICALL Java_org_openscenegraph_osg_db_ReadFile_nativeReadImageFile(JNIEnv *env, jclass, jstring filename)
+{
+	//__android_log_write(ANDROID_LOG_INFO,LOG_TAG,jstring2string(env,filename).c_str());
+	osg::Image* node = osgDB::readImageFile(jstring2string(env,filename));
+	if(node == 0)
+	{
+		LOGE("Error loading image");
+		return 0;
+	}
+	node->ref();
+	return reinterpret_cast<jlong>(node);
+
+}
+
+JNIEXPORT jboolean JNICALL Java_org_openscenegraph_osg_db_WriteFile_nativeWriteNodeFile(JNIEnv* env, jclass, jstring filename, jlong node_cptr)
+{
+	#define LOG_TAG "org.openscenegraph.osg.db"
```

Comment:
hm - somewhat can remember that I had to do this somewhat messy define here - but I don't know why anymore. I'll try to remove it and see if it still works.